pkg/registry/core/service/storage/rest.go

```diff
@@ -884,9 +895,12 @@ func (rs *REST) tryDefaultValidateServiceClusterIPFields(service *api.Service) e
 			break
 		}
 
-		// we have previously validated for ip correctness and if family exist it will match ip family
-		// so the following is safe to do
-		isIPv6 := netutil.IsIPv6String(ip)
+		parsedIP := net.ParseIP(ip)
```

Comment:
from the top of my head, I' m afraid that a wrong service with one valid IP and an invalid one, allocates the valid and fails later.

ClusterIPs: ["192.168.0.2","INVALID"]

Reply:
ic , we normalize and then parse, if something wrong parsing, return and validation catchs it later
:+1: